effect.cpp

```diff
@@ -776,16 +776,21 @@ static optional<ViewId> getProjectile(const DirEffectType& effect) {
   }
 }
 
-void applyDirected(WCreature c, Vec2 direction, const DirEffectType& type) {
+void applyDirected(WCreature c, Vec2 direction, const DirEffectType& type, optional<FXName> fx) {
   auto begin = c->getPosition();
   int range = type.getRange();
   for (Vec2 v = direction; v.length8() <= range; v += direction)
     if (!c->getPosition().plus(v).canEnterEmpty(MovementType({MovementTrait::FLY, MovementTrait::WALK}))) {
       range = v.length8();
       break;
     }
-  if (auto projectile = getProjectile(type))
+
+  if (fxesAvailable() && fx) {
```

Comment:
I think fxesAvailable() will be checked further on when rendering this effect, so no need to check here?

Reply:
The thing is, if fxes are disabled, we would like to show sprite based projectiles. That's why this check is required.